### PR TITLE
Testing/domUpdates/Refactoring

### PR DESCRIPTION
**Pull request by NakiNorton**
### Type of change made:
- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] Styling- no new features
- [X] Refactoring
- [ ] Planning

### Detailed Description
- Fixes bugs that resulted after our last merge
- Adds functionality to generate a random user on window load
- Adds some styling to input fields and new text to the steps info display
- Completes testing for domUpdates using spies
- Makes a start to improve consistency throughout code with use of single-quotes, and semi-colons

### Why is this change required? What problem does it solve?
- Fixes bugs that were stopping the display working correctly

### Were there any challenges that arose while implementing this feature? If so, how were the addressed?
- Lots of challenges. It took a long time to find what was causing the bugs. I pulled used an older branch to check what recent changes could've caused the issues.

### Files modified:
- [X] index.html
- [X] .scss // file name(s):
- [X] classes // file name(s):
- [X] test // files name(s):
- [X] scripts.js
- [X] Other files:

### CODE REVIEW

File: src/ActivityRepository.js [hunk 1]
```diff
@@ -42,16 +42,16 @@ class ActivityRepository {
   }
 
   checkIfStepGoalAchieved(date, user) {
-    let todaysRecord = this.individualEntryRecords.find(record => record.date === date)
-    let goalReached = (todaysRecord.numSteps >= user.dailyStepGoal)
+    let todaysRecord = this.individualEntryRecords.find(record => record.date === date);
```
Comment (leighlars) [inline src/ActivityRepository.js:45]: love the semi colon consistency

File: src/UserRepository.js [hunk 2]
```diff
@@ -131,27 +130,27 @@ class UserRepository {
...
-    let worstSleepers = this.users.filter(user => user.sleepInfo.findLastNightsSleepQual(this.todaysDate) < 3);
-    let sortedWorst =  worstSleepers.sort((a, b) => a.sleepQuality - b.sleepQuality);
-    if (sortedWorst) {
-      return sortedWorst.shift().name;    
-    } else {
-      return "n/a"
-    }
-  }
+  // findBestSleeper() {
+  //   let bestSleepers = this.users.filter(user => user.sleepInfo.findLastNightsSleepQual(this.todaysDate) > 3);
```
Comment (leighlars) [inline src/UserRepository.js:134]: good on you for deleting this

File: src/css/_display.scss [hunk 3]
```diff
@@ -95,12 +95,18 @@ main {
...
+  margin-bottom: .2vw;
 }
 
 .hydration-weekly-avg,
 .hydration-week-data-list {
   text-align: center;
   width: 100%;
 }
+
+.trend-line {
```
Comment (leighlars) [inline src/css/_display.scss:108]: I think we can actually delete this

File: src/domUpdates.js [hunk 4]
```diff
@@ -9,18 +9,16 @@ const domUpdates = {
...
+    this.displayName();
     this.stepCardDisplay();
     this.sleepCardDisplay();
     this.hydrationCardDisplay();
     this.stairsCardDisplay();
   },
   
   displayName() {
-    document.querySelector('#header-name').innerText = `${this.currentUser.getFirstName()}'S `;
+    document.querySelector('#header-name').innerText = `${this.currentUser.getFirstName()}'S FITLIT`;
```
Comment (leighlars) [inline src/domUpdates.js:21]: ooh yeah get rid of that span!!!

File: src/domUpdates.js [hunk 5]
```diff
@@ -38,111 +36,78 @@ const domUpdates = {
 
   showLeaderBoard() {
     let dropdownFriendsStepsContainer = document.querySelector('#dropdown-friends-steps-container');
-    let friendsStepsParagraphs = document.querySelectorAll('.friends-steps');
+    document.querySelectorAll('.friends-steps');
```
Comment (leighlars) [inline src/domUpdates.js:39]: I actually don't even think we need this query selector if we aren't using it

File: src/ActivityRepository.js [hunk 6]
```diff
@@ -68,24 +68,13 @@ class ActivityRepository {
...
-  }
-
   compareUserGoalWithCommunityGoal(userGoal, userRepo) {
-    let communityStepGoal = userRepo.getCommunityAvgStepGoal()
+    let communityStepGoal = userRepo.getCommunityAvgStepGoal();
     let goalDifference = userGoal - communityStepGoal;
-    if (goalDifference) { 
-      return `Your goal is ${goalDifference} steps above average!`
+    if (goalDifference > 0) { 
+      return `Your goal is ${goalDifference} steps above average!`;
```
Comment (Leasw144) [inline src/ActivityRepository.js:75]: Good work combing through the docs for inconsistencies. I think moving forward on other projects, norming on style conventions would be important

**Comment by Leasw144** [comment]
Looks good, thanks for getting the bug fixes handled

**Pull request merged**